core/tests/webdriverio_utils/ContributorDashboardAdminPage.js

```diff
@@ -232,7 +234,94 @@ var ContributorDashboardAdminPage = function () {
       `.e2e-test-language-selector-option=${language}`
     );
     await action.click(`${language} option selector`, selectorOption);
-    await action.click('Language Selector', languageSelector);
+  };
+
+  this.selectDate = async function (datePickerToggle, day, month, year) {
+    await action.click('Date Picker Toggle', $(datePickerToggle));
+
+    if (year) {
+      var nextMonthButton = $('.mat-calendar-next-button');
+      var prevMonthButton = $('.mat-calendar-previous-button');
+
+      var yearsToNavigate = year - new Date().getFullYear();
+
+      if (yearsToNavigate > 0) {
+        for (let i = 0; i < yearsToNavigate * 12; i++) {
+          await action.click('Next Month Button', nextMonthButton);
+        }
+      } else if (yearsToNavigate < 0) {
+        for (let i = 0; i < Math.abs(yearsToNavigate) * 12; i++) {
+          await action.click('Previous Month Button', prevMonthButton);
+        }
+      }
+    }
+
+    if (month) {
+      var nextMonthButton = $('.mat-calendar-next-button');
+      var prevMonthButton = $('.mat-calendar-previous-button');
+
+      var currentMonth = new Date().getMonth() + 1;
+      var monthsToNavigate = month - currentMonth;
+
+      if (monthsToNavigate > 0) {
+        for (let i = 0; i < monthsToNavigate; i++) {
+          await action.click('Next Month Button', nextMonthButton);
+        }
+      } else if (monthsToNavigate < 0) {
+        for (let i = 0; i < Math.abs(monthsToNavigate); i++) {
+          await action.click('Previous Month Button', prevMonthButton);
+        }
+      }
+    }
+
+    if (day) {
+      var daySelector = $(`aria/${day}`);
```

Comment:
I don't think you need to calculate the row and column to use chained selectors, since selectors in the chain do not need to have direct parent-child relationships in the DOM (though the example in our wiki happens to have such a direct relationship). According to the [webdriverio docs](https://webdriver.io/docs/selectors/#chain-selectors), each selector in the chain begins searching at the preceding element in the chain. The search is still recursive though. I think you can use an identifier for the whole calendar (or an enclosing element) for the first selector in the chain and your aria selector for the second selector

Reply:
@U8NWXD, I tried adding `e2e-*` to a class of an ancestor to the desired node. However, whenever date picker is toggled an invisible overlay covers entire screen on with date picker in displayed, and the ancestor to this is the body tag itself in the DOM.
![Screenshot (151)](https://github.com/oppia/oppia/assets/92575005/3693f0b9-4387-4d5c-b678-0ce1bf5e6de4)
PTAL

Reply:
Okay, in this case I agree that you won't be able to use an `e2e-*` class. I still think `aria/${day}` is too broad though. Can you use a chained selector where the first selector identifies the date picker calendar (e.g. by a class, even if that class isn't of the form `e2e-*`), and the second selector is `aria/${day}`? That way we only have to assume that `aria/${day}` uniquely identifies an element in the date picker rather than assuming that it identifies a unique element on the whole page (which is more likely to change in the future)

Reply:
Sure, this sounds good.

Reply:
done